### PR TITLE
Upgrade from nose, which is no longer maintained, to pytest

### PR DESCRIPTION
**Pull request by bschoening**
This is a minimalist upgrade to use the pytest framework.  It uses backward compatibility with nose to maintain nose imports.

Several tests had to be updated or fixed, which seemed unrelated to the pytest upgrade.  With these changes, pytest runs a clean set of tests.

% pytest
========================================================================== test session starts ===========================================================================
platform darwin -- Python 3.9.10, pytest-7.0.0, pluggy-1.0.0
rootdir: /Users/brad/mysto/cassandra/pylib
plugins: anyio-2.1.0
collected 87 items                                                                                                                                                       

cqlshlib/test/test_constants.py .                                                                                                                                  [  1%]
cqlshlib/test/test_copyutil.py .                                                                                                                                   [  2%]
cqlshlib/test/test_cql_parsing.py .......................                                                                                                          [ 28%]
cqlshlib/test/test_cqlsh_completion.py ..................                                                                                                          [ 49%]
cqlshlib/test/test_cqlsh_output.py ...................................                                                                                             [ 89%]
cqlshlib/test/test_sslhandling.py .....                                                                                                                            [ 95%]
cqlshlib/test/test_unicode.py ....                                                                                                                                 [100%]

============================================================== 87 passed, 15 warnings in 319.37s (0:05:19) ===============================================================

### CODE REVIEW

**Comment by bschoening** [comment]
I need to fix the hard coded credentials in PlainTextAuthProvider(username='cassandra', password='cassandra')

File: pylib/cassandra-cqlsh-tests.sh [hunk 1]
```diff
@@ -126,14 +118,14 @@ ccm start --wait-for-binary-proto
...
-nosetests
+pytest
 RETURN="$?"
 
 ccm remove
 # hack around --xunit-prefix-with-testsuite-name not being available in nose 1.3.7
-sed -i "s/testsuite name=\"nosetests\"/testsuite name=\"${TESTSUITE_NAME}\"/g" nosetests.xml
-sed -i "s/testcase classname=\"cqlshlib./testcase classname=\"${TESTSUITE_NAME}./g" nosetests.xml
-mv nosetests.xml ${WORKSPACE}/cqlshlib.xml
+# sed -i "s/testsuite name=\"nosetests\"/testsuite name=\"${TESTSUITE_NAME}\"/g" nosetests.xml
```
Comment (bereng) [inline pylib/cassandra-cqlsh-tests.sh]: Why not simply remove these lines?

Comment (bschoening) [reply pylib/cassandra-cqlsh-tests.sh]: Yes, I will do that.

Comment (bschoening) [reply pylib/cassandra-cqlsh-tests.sh]: @bereng this has been updated

**Comment by smiklosovic** [comment]
https://github.com/apache/cassandra/commit/db3583318258b14b27abe1529f8893da2ff22bf5

File: pylib/cassandra-cqlsh-tests.sh (hunk 1 above)
Comment (bereng) [reply pylib/cassandra-cqlsh-tests.sh]: Thx!

**Pull request closed without merging**